core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -168,11 +227,18 @@ private String addLibraryDetails(Bundle bundle, String bomText) {
                     LOG.warn("Unexpected syntax for {} (expected Map), ignoring", BROOKLYN_CATALOG);
                 }
             }
-            final String updatedBom = new Yaml().dump(bom);
+            final String updatedBom = backToYaml(bom);
```

Comment:
Something that I didn't consider before when discussing this approach is that parsing and re-serializing the yaml will change its formatting. Not a problem for Brooklyn internals, but will be surprising to users when viewing the plan in the catalog. Not to be addressed in this PR, but we should keep it in mind.